pkg/csi/controller_server.go

```diff
@@ -89,11 +116,30 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 			Name:      req.Name,
 		},
 		Spec: corev1.PersistentVolumeClaimSpec{
-			StorageClassName: pointer.StringPtr(cs.hostStorageClass),
-			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany},
-			VolumeMode:       &volumeMode,
+			AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany},
+			VolumeMode:  &volumeMode,
 		},
 	}
+
+	/*
+	 * Let's handle SC
+	 * SC define > controller define
+	 *
+	 * Checking mechanism:
+	 * 1. If guest cluster is not be allowed to list host cluster StorageClass, just continue.
+	 * 2. If guest cluster can list host cluster StorageClass, check it.
+	 */
+	if val, exists := volumeParameters[paramHostSC]; exists {
+		targetSC = val
+	}
+	if targetSC != "" {
+		if valid := cs.isStorageClassValid(targetSC); !valid {
+			return nil, status.Error(codes.NotFound, "Associated StorageClass does not exist.")
+		}
+		pvc.Spec.StorageClassName = pointer.StringPtr(targetSC)
+		logrus.Infof("Set up the target StorageClass to : %s", *pvc.Spec.StorageClassName)
+	}
```

Comment:
Can we move this into the `exists` block?

Reply:
Done! Thanks for the tips!